src/Routes.tsx

```diff
@@ -57,159 +62,239 @@ export interface ChildProps {
-      <AuthenticatedRoute
+      <Route
         path="/training/:id"
         exact
-        component={ExperienceModule}
-        props={childProps}
+        {...rest}
+        element={
+          <AuthenticatedRoute>
+            <ExperienceModule {...childProps} />
+          </AuthenticatedRoute>
+        }
       />
-      <AppliedRoute
+      <Route
         path="/chat/:id"
         exact
-        component={Room}
-        props={childProps}
+        {...rest}
+        /* @ts-ignore */
+        element={<Room {...childProps} />}
+        // props={childProps}
       />
-      <AuthenticatedRoute
+      <Route
         path="/settings/password"
         exact
-        component={ChangePassword}
-        props={childProps}
+        {...rest}
+        element={
+          <AuthenticatedRoute>
+            <ChangePassword {...childProps} />
+          </AuthenticatedRoute>
+        }
       />
-      <AuthenticatedRoute
+      <Route
         path="/profile/:id"
         exact
-        component={Profile}
-        props={childProps}
+        {...rest}
+        element={
+          <AuthenticatedRoute>
+            {/* @ts-ignore */}
+            <Profile {...childProps} />
+          </AuthenticatedRoute>
+        }
       />
-      <AuthenticatedRoute
+      <Route
         path="/profile/edit/:id"
         exact
-        component={EditProfile}
-        props={childProps}
+        {...rest}
+        element={
+          <AuthenticatedRoute>
+            <EditProfile {...childProps} />
+          </AuthenticatedRoute>
+        }
       />
-      <AuthenticatedRoute
+      <Route
         path="/profile/languages/:id"
         exact
-        component={LanguageSelector}
-        props={childProps}
+        {...rest}
+        element={
+          <AuthenticatedRoute>
+            <LanguageSelector {...childProps} />
+          </AuthenticatedRoute>
+        }
       />
-      <AppliedRoute
+      <Route
         path="/onboarding/user"
         exact
-        component={CreateUser}
-        props={childProps}
+        {...rest}
+        element={<CreateUser {...childProps} />}
       />
-      <AuthenticatedRoute
+      <Route
         path="/sandbox"
         exact
-        component={Sandbox}
-        props={childProps}
+        {...rest}
+        element={
+          <AuthenticatedRoute>
+            <Sandbox {...childProps} />
+          </AuthenticatedRoute>
+        }
       />
-      <AuthenticatedRoute
+      <Route
         path="/mentorship"
         exact
-        component={MentorDashboard}
-        props={childProps}
+        {...rest}
+        element={
+          <AuthenticatedRoute>
+            <MentorDashboard {...childProps} />
+          </AuthenticatedRoute>
+        }
       />
-      <AuthenticatedRoute
+      <Route
         path="/mentorship/:id"
         exact
-        component={Profile}
-        props={childProps}
+        {...rest}
+        element={
+          <AuthenticatedRoute>
+            {/* @ts-ignore */}
```

Comment:
Why do we have these ts-ignores here? Can you post the message here in this comment?

Reply:
Type '{ reviewMode: boolean; isAuthenticated: boolean; userHasAuthenticated: (b: boolean) => void; user: User; setLoading: (b: boolean) => void; connectSocket: (id?: string | undefined) => Promise<...>; ... 12 more ...; coaches: Coach[]; }' has no properties in common with type 'IntrinsicAttributes'

Also ran into import { RouteComponentProps } from ".pnpm/@types+react-router@5.1.18/node_modules/@types/react-router"; which no longer exists for react router 6, so I need to get rid of those as well.

Reply:
Yeah, you should remove that types file / update it. Delete `node_modules` and run a fresh install to remove any cached npm modules that could be the cause of it.